docs/docs/building.md

```diff
@@ -4,158 +4,208 @@ title: Building an upgradeable application
 
-After running this command, the new versions of your project's contracts are deployed in the blockchain. 
-However, the already deployed proxies are still running with the old implementations. You need to upgrade
-each of the proxies individually. To do so, you just need to run this for every contract: 
+The first time you run this command for a specific network, a new
+`package.zos.<network>.json` will be created. This file will reflect the status
+of your project in that network.
 
-```
-zos upgrade <contract_name_1> --network <network>
-zos upgrade <contract_name_2> --network <network>
-...
-zos upgrade <contract_name_n> --network <network>
-```
+TODO: test this and comment about the resulting logs. On the published zos, the set-stdlib command fails, wait for a new release.
 
-In our simple example:
-```
-zos upgrade MyContract --network development
-```
+## Upgrading a contract
 
-Voilà! Your contract has now been upgraded. The address is the same as before, but the code has been changed to the latest version. Repeat the same steps for every code update you want to perform.
+The rules for our basil lights will now be set in stone, enforced by the immutability of the Ethereum blockchain. This sounded great... until we found an embarrassing bug: we are never updating `highestDonation`! The donations would not grow as we expect because the stakes will never be higher than 10 wei. Luckily we caught this before going to mainnet, but are we sure that our contract is bullet proof and something like this will never happen again? The same feature that makes Ethereum secure is now making us feel insecure about our programming abilities and scared of finding a security vulnerability too late.
+
+Fear not, ZeppelinOS allows us to keep the transparency and immutability of a deployed version of a contract, and also to opt-in for a contract in which the owner can upgrade the implementation. This is done through a proxy that forwards the calls to the latest implementation of the contract. To create a proxy for Basil, run:
+
+    zos create Basil --network development --init --args $OWNER
+
+Let's fix our bug. Edit `contracts/Basil.sol` to add the missing line to the `donate` function:
+
+    function donate(uint256 _red, uint256 _green, uint256 _blue) public payable {
+      require(_red < 256);
+      require(_green < 256);
+      require(_blue < 256);
+      require(msg.value > highestDonation);
+      red = _red;
+      green = _green;
+      blue = _blue;
+      highestDonation = msg.value;
+      emit NewDonation(msg.sender, msg.value, red, green, blue);
+    }
+
+And to finish our fix, we compile the patched contract, sync with ZeppelinOS and upgrade the proxy:
+
+    zos upgrade Basil <proxy_address_1> --network development
+
+TODO: where does the proxy address comes from?
```

Comment:
same as above.

Reply:
removed. I'll test the commands now and then come back to explain this arg.